### PR TITLE
fixed compiler swithed of zcee

### PR DESCRIPTION
**Pull request by Liaoshihua**

### CODE REVIEW

File: gcc/config/riscv/zce.md [hunk 1]
```diff
@@ -22,7 +22,7 @@
 (define_insn "*zero_extendsidi2_zcee"
   [(set (match_operand:DI 0 "register_operand" "=r,r")
 	(zero_extend:DI (match_operand:SI 1 "nonimmediate_operand" "r,m")))]
-  "TARGET_64BIT && TARGET_ZCEE"
+  "TARGET_64BIT && TARGET_ZCEE && TARGET_ZEXT"
```
Comment (pz9115) [inline gcc/config/riscv/zce.md]: If we just use option -march=zcee, the instruction may not gen, so I think these better use || but not && TARGET_ZEXT

File: gcc/testsuite/gcc.target/riscv/zcee-sext.c [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /* { dg-do compile { target { riscv64*-*-* } } } */
-/* { dg-options "-march=rv64gc_zcee -mabi=lp64 -O2" } */
+/* { dg-options "-march=rv64gc_zcee -mabi=lp64 -mzce-sext -O2" } */
```
Comment (pz9115) [inline gcc/testsuite/gcc.target/riscv/zcee-sext.c]: Same problem, If we need to alway use -mzce-sext to enable instructions or only need zcee in march

File: gcc/config/riscv/riscv.opt [hunk 3]
```diff
@@ -133,10 +133,10 @@ Take advantage of linker relaxations to reduce the number of instructions
...
 
 mzce-sext
-Target Bool Var(riscv_mzce_sext) Init(0)
+Target RejectNegative Mask(SEXT)
 
 mzce-zext
-Target Bool Var(riscv_mzce_zext) Init(0)
+Target RejectNegative Mask(ZEXT)
 
 mzce-cmul
```
Comment (pz9115) [inline gcc/config/riscv/riscv.opt:141]: Sense there no c.mul will occur in gcc patch, I think these option is not  nesscessary, but you can just pass it to binutils use in as part.

File: gcc/config/riscv/zce.md (hunk 1 above)
Comment (Liaoshihua) [reply gcc/config/riscv/zce.md]: I think we must always use mzce-sext and mzce-zext to start sext.* and zext.*. 
If we just only use -march=zcee,the option can not transmit to binutils.
Without mzce-sext in binutils , sext.* zext.* only can not be used .

**Comment by Liaoshihua** [comment]
/riscv64-unknown-elf-gcc -mzce-sext -mzce-zext -march=rv64gc -O2 -o zce.elf  zceetest.c 
image
Correct assembly can be generated without specifying -march = rv64gc_zcee using only -mzce-sext -mzce-zext.
But after objdump,the result just like the picture:
image

**Pull request closed without merging**